test/demo/connection.inttest.ts

```diff
@@ -171,3 +232,9 @@ async function hasEventListeners(identity: Identity, eventType: ChannelEvents['t
     }, eventType);
     return identities.some(id => id.name === identity.name && id.uuid === identity.uuid);
 }
+
+async function isRegistered(identity: Identity): Promise<boolean> {
```

Comment:
My feeling is no.  This and the similar functions in this file are all kind cheeky in the context of integration tests. Rather than looking at the API output, we're looking at the internal state of the provider, which is dodgy, but kinda necessary in these tests. I wouldn't want to communicate "this is a thing we should be doing" by putting it in a common util file